storm-server/src/main/java/org/apache/storm/scheduler/resource/strategies/scheduling/IStrategy.java

```diff
@@ -22,9 +23,10 @@
  * Scheduler should call {@link #prepare(Map)} followed by {@link #schedule(Cluster, TopologyDetails)}.
  * <p>
  *     A fully functioning implementation is in the abstract class {@link BaseResourceAwareStrategy}.
- *     Subclasses classes should extend {@link BaseResourceAwareStrategy#BaseResourceAwareStrategy()}
- *     in their constructors as in {@link GenericResourceAwareStrategy}, {@link DefaultResourceAwareStrategy})
- *     and {@link ConstraintSolverStrategy}.
+ *     Subclasses classes should extend
+ *     {@link BaseResourceAwareStrategy#BaseResourceAwareStrategy(boolean, BaseResourceAwareStrategy.NodeSortType)}
+ *     in their constructors (as in {@link GenericResourceAwareStrategy}, {@link DefaultResourceAwareStrategy}
+ *     and {@link ConstraintSolverStrategy}).
```

Comment:
The javadoc comment should be reverted - signature of extended classes do not have the params.